datafusion/physical-expr/src/functions.rs

```diff
@@ -191,6 +191,30 @@ pub(crate) enum Hint {
     AcceptsSingular,
 }
 
+/// A helper function used to infer the length of arguments of Scalar functions and convert
+/// [`ColumnarValue`]s to [`ArrayRef`]s with the inferred length. Note that this function
+/// only works for functions that accept either that all arguments are scalars or all arguments
+/// are arrays with same length. Otherwise, it will return an error.
+pub fn process_scalar_func_inputs(args: &[ColumnarValue]) -> Result<Vec<ArrayRef>> {
```

Comment:
What would you think about naming this something more descriptive of what it does rather than what it is used for? 

Something like this perhaps:

```suggestion
pub fn columnar_values_to_array(args: &[ColumnarValue]) -> Result<Vec<ArrayRef>> {
```